### PR TITLE
feat: add tvOS support

### PR DESCRIPTION
**Pull request by mikehardy**
### Description

This supercedes:
- #8109

It is based on that PR from @adamkoch but I can't push to that branch and I have done a series of little fixups on it

### Related issues

Discussion:
- https://github.com/invertase/react-native-firebase/discussions/5722#discussioncomment-11125191

### Release Summary

This should be a squash merge, the PR title should serve for release as conventional commit message

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

The base of the PR style-wise is adding the platform to package.json then using it in podspecs, that is tested already by library consumer so should work

The compile ifdefs will be tested on iOS at least and that should be sufficient

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/auth/ios/RNFBAuth/RNFBAuthModule.m [hunk 1]
```diff
@@ -55,7 +55,7 @@
 static __strong NSMutableDictionary *emulatorConfigs;
 // Used for caching credentials between method calls.
 static __strong NSMutableDictionary<NSString *, FIRAuthCredential *> *credentials;
-#if !TARGET_OS_TV
+#if TARGET_OS_IOS
```
Comment (mikehardy) [inline packages/auth/ios/RNFBAuth/RNFBAuthModule.m:58]: all the multifactor stuff is guarded this way, example: https://github.com/firebase/firebase-ios-sdk/blob/ae2554220afb03a5d6fc389c28a86061a12c7caa/FirebaseAuth/Sources/ObjC/FIRMultiFactorConstants.m#L18

File: packages/app-check/ios/RNFBAppCheck/RNFBAppCheckProvider.m [hunk 2]
```diff
@@ -55,18 +55,18 @@ - (void)configure:(FIRApp *)app
   }
 
   if ([providerName isEqualToString:@"appAttest"]) {
-    if (@available(iOS 14.0, macCatalyst 14.0, tvOS 15.0, watchOS 9.0, *)) {
+    if (@available(iOS 14.0, macOS 11.0, macCatalyst 14.0, tvOS 15.0, watchOS 9.0, *)) {
```
Comment (mikehardy) [inline packages/app-check/ios/RNFBAppCheck/RNFBAppCheckProvider.m:58]: just an oversight before, min versions guided by https://firebase.google.com/docs/ios/learn-more#firebase_library_support_by_platform

File: packages/app-check/ios/RNFBAppCheck/RNFBAppCheckProvider.m [hunk 3]
```diff
@@ -55,18 +55,18 @@ - (void)configure:(FIRApp *)app
...
-      DLog(@"AppAttest unavailable: it requires iOS14+, macCatalyst14+ or tvOS15+. Installing "
+      DLog(@"AppAttest unavailable: it requires iOS14+, macOS 11+, macCatalyst14+ or tvOS15+. Installing "
            @"debug provider to guarantee invalid tokens in this invalid configuration.");
       self.delegateProvider = [[FIRAppCheckDebugProvider alloc] initWithApp:app];
     }
   }
 
   if ([providerName isEqualToString:@"appAttestWithDeviceCheckFallback"]) {
-    if (@available(iOS 14.0, *)) {
+    if (@available(iOS 14.0, macOS 11.0, macCatalyst 14.0, tvOS 15.0, watchOS 9.0, *)) {
```
Comment (mikehardy) [inline packages/app-check/ios/RNFBAppCheck/RNFBAppCheckProvider.m:70]: this should have been guarded for all the platforms during initial implementation, oversight

File: packages/app/package.json [hunk 4]
```diff
@@ -77,7 +77,7 @@
       "firebase": "11.4.0",
       "iosTarget": "13.0",
       "macosTarget": "10.15",
-      "tvosTarget": "12.0"
+      "tvosTarget": "13.0"
```
Comment (mikehardy) [inline packages/app/package.json:80]: FirebaseAuth is currently the highest minimum version for iOS, consulted it here and it was 13 for tvOS as well
https://github.com/firebase/firebase-ios-sdk/blob/36da449b15619d301302faeba639c7893d576220/FirebaseAuth.podspec#L24

Which matches Core, so I chose 13
https://github.com/firebase/firebase-ios-sdk/blob/36da449b15619d301302faeba639c7893d576220/FirebaseCore.podspec#L23

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**